### DISCUSSION TITLE
Comparing profiles

### DISCUSSION THREAD

**Discussion by cyriltovena** [grafana/phlare#295] [Polls]
Other product often offer a way to compare flamegraph/profiles across different range of time, at Grafana we're not heavy user of this feature (yet) but we would love to hear how useful this has been for others.

**Comment by kami619** [comment]
@cyriltovena this would be a really good enabler to shift the profiling of applications to the left. There are very few tools available in the performance engineering community mostly revolving around this [article](https://www.brendangregg.com/blog/2014-11-09/differential-flame-graphs.html) to create the diff between two given CPU flamegraphs. So it would be a great benefit to start there and expand on compare two snapshots of profiles of an application and provide a summary of information.

**Comment by agnivade** [comment]
Hi @cyriltovena - I work at [Mattermost](https://github.com/mattermost/mattermost-server) and we are looking at a continuous profiling solution currently that will allow us to compare profiles and catch performance regressions in an automated manner. Currently, we manually compare profiles by hand which is a laborious process.

There are some tools out in the market that does this: https://pyroscope.io/. However, we use Grafana+Prom+Loki heavily in our stack and would greatly appreciate if phlare added the feature of comparing profiles. Then it would be very organic and would fit perfectly with everything else.

Otherwise, we will have to look for solutions elsewhere.

**Reply by cyriltovena** [reply]
Good news it's coming we're working on this. We're bringing the full Pyroscope UI to Grafana.
